### PR TITLE
Implemented basic frontend

### PR DESCRIPTION
**Pull request by christopherklaus**
ToDo: 
- [ ] tests

### CODE REVIEW

File: backend/Dockerfile.dev [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # base image
-FROM node:12-alpine
+FROM node:12
```
Comment (martinzuern) [inline backend/Dockerfile.dev:2]: pls revert

File: frontend/src/store/index.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+Vue.use(Vuex);
+
+export default new Vuex.Store({
+  state: {
+    game: {} as Game,
+    player: {} as Player
+  },
+  getters: {
+    apiFunction: (state) => (payload = {}) => {
+      const baseURL = 'http://localhost:3000/api/v1'
```
Comment (martinzuern) [inline frontend/src/store/index.ts:16]: Can this be injected using env vars?

Comment (christopherklaus) [reply frontend/src/store/index.ts:16]: Sure

File: backend/Dockerfile.dev (hunk 1 above)
Comment (christopherklaus) [reply backend/Dockerfile.dev:2]: Aye

File: frontend/src/helpers/api.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import axios from 'axios'
+
+export default() => {
```
Comment (martinzuern) [inline frontend/src/helpers/api.ts:3]: This won't throw if the environment variable is not set, and will likely cause trouble if we want to deploy it behind a proxy later on. 

Also, axios will set the headers automatically.

```suggestion
const baseURL = (new URL('/api/v1', process.env.VUE_APP_BACKEND_URL || window.location.origin)).toString();

export default() => {
  return axios.create({
    baseURL,
  })
}
```

File: frontend/src/router/index.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+Vue.use(VueRouter);
+
+const routes: RouteConfig[] = [
+  {
+    path: '/',
+    name: 'NewGame',
+    component: NewGame,
+  },
+  {
+    path: '/:gameId',
```
Comment (martinzuern) [inline frontend/src/router/index.ts:15]: how about being restful?

```suggestion
    path: '/games/:gameId',
```

File: frontend/src/store/index.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  mutations: {
+    setGame(state, game) {
+      state.game = game
+    },
+    setPlayer(state, player) {
+      state.player = player
+    },
+  },
+  actions: {
+    executeAPI: ({ state, getters, dispatch }, { action, payload }) => {
```
Comment (martinzuern) [inline frontend/src/store/index.ts:42]: I guess it's hart to type this using this code, but that could be done in a separate PR. :)

File: frontend/src/views/InGame.vue [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      return this.$store.state.game
+    },
+    player() {
+      return this.$store.state.player
+    },
+  },
+  mounted() {
+    if (localStorage.token) {
+      // const websocket = io.connect(process.env.VUE_APP_BACKEND_URL)
+      const websocket = io.connect(process.env.VUE_APP_BACKEND_URL, { autoConnect: false })
```
Comment (martinzuern) [inline frontend/src/views/InGame.vue:34]: Here, we need the same logic as in https://github.com/martinzuern/cards-offending-mankind/pull/14/files#diff-86e7e8d66da8191a525c19347e6ff535R3

**Pull request merged**